src/DwitTech.AccountService.Data/repository/UserRepository.cs

```diff
@@ -1,20 +1,63 @@
 ﻿using DwitTech.AccountService.Data.Context;
 using DwitTech.AccountService.Data.Entities;
-using System;
-using System.Collections.Generic;
+using Microsoft.EntityFrameworkCore;
 using System.Linq;
-using System.Text;
-using System.Threading.Tasks;
+using static DwitTech.AccountService.Data.Repository.IUserRepository;
 
 namespace DwitTech.AccountService.Data.Repository
 {
-    
-
-    
     public class UserRepository : IUserRepository
     {
-        
+        private readonly AccountDbContext _dbContext;
+
+        public UserRepository(AccountDbContext dbContext)
+        {
+            _dbContext = dbContext;
+        }
+
+        public async Task<ValidationCode> GetActivationDetail(string activationCode)
+        {
+            return await _dbContext.ValidationCodes.Where(x => x.Code == activationCode).FirstOrDefaultAsync();
+           
+        }
+
+        public async Task<bool> GetUserStatus(int id)
+        {
+            var status = await _dbContext.Users.Where(x => x.Id == id).Select(x => x.Status).FirstOrDefaultAsync();
+            if (status == UserStatus.Inactive)
+            {
+                return true;
+            }
+            return false;
+        }
+
+        public async Task<bool> ValidateActivationCodeExpiry(string activationCode)
+        {
+            var validationCode = await GetActivationDetail(activationCode);
+
+            DateTime ExpiredTime = validationCode.CreatedTime.AddMinutes(10);           
```

Comment:
variable should begin with lowercase. That being said, I think this should be called something in the likeness of "activeDuration". expiredTime doesn't seem to fit in this context.

Reply:
> variable should begin with lowercase.

Resolved